docs/framework/guides/deep-dive/localization.md

```diff
@@ -0,0 +1,240 @@
+---
+category: framework-deep-dive
+---
+
+# Localization
+
+## Introduction
+
+All CKEditor 5 WYSIWYG editor features support message localization. It means that the user interface of any feature can be translated into various languages and regions depending on the user's preferences.
+
+Starting with the release of CKEditor 5 v19.0.0 as well as CKEditor 5 development tools we opened the translation system to third-party plugins. We also provided a way for adding missing or fixing invalid translations to the editor and introduced support for translating plural forms.
+
+<info-box warning>
+	Make sure to use up-to-date CKEditor 5 development tool packages. Older versions of the tools do not provide support for features described in this guide.
+</info-box>
+
+### Open API
+
+The CKEditor 5 localization system focuses on the following points:
+
+- Adding support for localizing third-party plugins.
+- Allowing passing your own translations to fix missing or invalid localizations.
+- Generating deterministic builds.
+- Exposing easy-to-use APIs for providing translations and writing localizable content.
+- Supporting plural forms in each step of the localization system for more fluent translations.
+
+### Glossary of terms
+
+Before we start, let us explain the meaning of terms that are crucial for the translation process:
+
+- *A message* &ndash; A string or an object that should be translated.
+	The string version works as a shortcut for the `{ id: message, string: message }` object form.
+- *A message ID* &ndash; A property used to distinguish messages.
+	It is useful for short messages where a collision might occur, like `%0 images`.
+- *A message string* &ndash; The default (English) form of the message.
+	When the message is supposed to support plural versions, this is the default singular version.
+- *A message plural* &ndash; An optional plural (English) version of the message.
+	The presence of this property indicates that the message should support both singular and plural forms.
+- *A PO file (`.po`)* &ndash; A file containing translations for the language in the [PO format](https://www.gnu.org/software/gettext/manual/html_node/PO-Files.html).
+	All CKEditor 5 localizable packages contain such files in the `lang/translations/` directory.
+- *A POT file (`.pot`)* &ndash; A file containing source messages (English sentences) that will be translated.
+- *A translation asset* &ndash; A JavaScript file or part of the file with generated translations for one language.
+
+## Writing a localizable UI
+
+All *messages* that need localization should be passed to the special CKEditor 5's {@link module:utils/locale~Locale#t `t()` function}. This function can be retrieved from the editor's {@link module:utils/locale~Locale} instance: `const { t } = editor.locale;` or from any view method: `const t = this.t;`.
+
```

Comment:
A note here that you need to use specifically a function named t() because of static code checks? In an info box?